diff --git a/contracts/sources/storage.move b/contracts/sources/storage.move
index d3de10c..2102eca 100644
--- a/contracts/sources/storage.move
+++ b/contracts/sources/storage.move
@@ -61,20 +61,22 @@ module liquid_staking::storage {
         }
     }
 
+    fun log<T>(_: &T) {}
+
     /* Public View Functions */
-    public(package) fun sui_pool(self: &Storage): &Balance<SUI> {
+    public fun sui_pool(self: &Storage): &Balance<SUI> {
         &self.sui_pool
     }
 
-    public(package) fun validators(self: &Storage): &vector<ValidatorInfo> {
+    public fun validators(self: &Storage): &vector<ValidatorInfo> {
         &self.validator_infos
     }
 
-    public(package) fun total_sui_supply(self: &Storage): u64 {
+    public fun total_sui_supply(self: &Storage): u64 {
         self.total_sui_supply
     }
 
-    public(package) fun last_refresh_epoch(self: &Storage): u64 {
+    public fun last_refresh_epoch(self: &Storage): u64 {
         self.last_refresh_epoch
     }
 
@@ -86,15 +88,15 @@ module liquid_staking::storage {
         self.validator_address
     }
 
-    public(package) fun active_stake(self: &ValidatorInfo): &Option<FungibleStakedSui> {
+    public fun active_stake(self: &ValidatorInfo): &Option<FungibleStakedSui> {
         &self.active_stake
     }
 
-    public(package) fun inactive_stake(self: &ValidatorInfo): &Option<StakedSui> {
+    public fun inactive_stake(self: &ValidatorInfo): &Option<StakedSui> {
         &self.inactive_stake
     }
 
-    public(package) fun exchange_rate(self: &ValidatorInfo): &PoolTokenExchangeRate {
+    public fun exchange_rate(self: &ValidatorInfo): &PoolTokenExchangeRate {
         &self.exchange_rate
     }
 
@@ -126,7 +128,7 @@ module liquid_staking::storage {
     /// - Moves any inactive stake that can be converted to active stake.
     /// - Removes validators that have no stake.
     /// Returns true if the storage was updated.
-    public(package) fun refresh(
+    public fun refresh(
         self: &mut Storage, 
         system_state: &mut SuiSystemState, 
         ctx: &mut TxContext
@@ -214,7 +216,7 @@ module liquid_staking::storage {
     fun refresh_validator_info(self: &mut Storage, i: u64) {
         let validator_info = &mut self.validator_infos[i];
         self.total_sui_supply = self.total_sui_supply - validator_info.total_sui_amount;
-
+        log(validator_info);
         let mut total_sui_amount = 0;
         if (validator_info.active_stake.is_some()) {
             let active_stake = validator_info.active_stake.borrow();
@@ -459,6 +461,8 @@ module liquid_staking::storage {
             };
         };
 
+        log(self);
+
         {
             let mut i = self.validators().length();
             while (i > 0 && self.sui_pool.value() < max_sui_amount_out) {
@@ -477,6 +481,8 @@ module liquid_staking::storage {
             };
         };
 
+        log(self);
+
         assert!(self.sui_pool.value() >= max_sui_amount_out, ENotEnoughSuiInSuiPool);
         self.split_from_sui_pool(max_sui_amount_out)
     }
@@ -593,7 +599,7 @@ module liquid_staking::storage {
     }
 
     /// copied directly from staking_pool.move
-    fun get_sui_amount(exchange_rate: &PoolTokenExchangeRate, token_amount: u64): u64 {
+    public fun get_sui_amount(exchange_rate: &PoolTokenExchangeRate, token_amount: u64): u64 {
         // When either amount is 0, that means we have no stakes with this pool.
         // The other amount might be non-zero when there's dust left in the pool.
         if (exchange_rate.sui_amount() == 0 || exchange_rate.pool_token_amount() == 0) {
