diff --git a/contracts/sources/storage.move b/contracts/sources/storage.move
index d3de10c..3b499ba 100644
--- a/contracts/sources/storage.move
+++ b/contracts/sources/storage.move
@@ -70,11 +70,11 @@ module liquid_staking::storage {
         &self.validator_infos
     }
 
-    public(package) fun total_sui_supply(self: &Storage): u64 {
+    public fun total_sui_supply(self: &Storage): u64 {
         self.total_sui_supply
     }
 
-    public(package) fun last_refresh_epoch(self: &Storage): u64 {
+    public fun last_refresh_epoch(self: &Storage): u64 {
         self.last_refresh_epoch
     }
 
@@ -437,6 +437,8 @@ module liquid_staking::storage {
         unstaked_sui_amount
     }
 
+
+    fun log<T>(_: &T) {}
     // This function approximately unstakes n SUI from validators, then returns up to n SUI.
     public(package) fun split_n_sui(
         self: &mut Storage,
@@ -459,6 +461,8 @@ module liquid_staking::storage {
             };
         };
 
+        log(self);
+
         {
             let mut i = self.validators().length();
             while (i > 0 && self.sui_pool.value() < max_sui_amount_out) {
@@ -477,6 +481,8 @@ module liquid_staking::storage {
             };
         };
 
+        log(self);
+
         assert!(self.sui_pool.value() >= max_sui_amount_out, ENotEnoughSuiInSuiPool);
         self.split_from_sui_pool(max_sui_amount_out)
     }
